source/qx/io/jsonrpc/message/Request.js

```diff
@@ -1,62 +0,0 @@
-qx.Class.define("qx.io.jsonrpc.message.Request",{
-  extend: qx.io.jsonrpc.Notification,
-  statics: {
-    /**
-     * Static counter for all request ids
-     */
-    __current_request_id : 0,
-
-    getCurrentId() {
-      return qx.io.jsonrpc.message.Request.__current_request_id;
-    },
-
-    resetId() {
-      qx.io.jsonrpc.message.Request.__current_request_id = 0;
-    }
-  },
-  properties: {
-    /**
-     * The integer id of the request
-     */
-    id : {
-      check: value => qx.lang.Type.isNumber(value) && parseInt(value) === value
-    },
-    /**
-     * Promise that is resolved when the response to this request has been received
-     */
-    promise: {
-      check: "qx.Promise"
-    }
-  },
-
-  /**
-   * JSON-RPC request constructor
-   * @param {String} method
-   * @param {Array|Object?} params
-   * @param {Number?} id
-   *    Optional integer id. If not provided, an auto-incremented id will be
-   *    used.
-   */
-  construct(method, params, id) {
-    this.base(arguments, method, params);
-    if (id === undefined) {
-      id = ++this.self(arguments).current_request_id;
```

Comment:
thanks, I hadn't thought of that. Why was this.self(arguments) ever used then?

Reply:
I have never encountered a case where `this.self(arguments)` does what I need, while `this.constructor` does not. I don't ever use the former; I use the latter frequently.

Fine. You made me go look at the source. :-)

Bootstrap.js does this, in method `extendClass`:
```
      construct.self = clazz.constructor = proto.constructor = clazz;
```
That, right there, has all of the information to access the static members.

In core/Object.js, the `self()` member method is defined:
```
    /**
     * Returns the static class (to access static members of this class)
     *
     * @param args {arguments} the arguments variable of the calling method
     * @return {var} the return value of the method of the base class.
     */
    self : function(args) {
      return args.callee.self;
    },
```
It turns out that this code isn't even allowed, any longer, in strict mode. `args.callee`, specifically.

`args.callee` is obtaining `construct` from above, I believe, so this is accessing  `construct.self` from above. 

I see no reason to ever use `this.self(arguments)` in preference to `this.constructor`.

Reply:
Maybe we should deprecate it then, and remove it from the documentation? It would be great if you could create an issue for this.

Reply:
I think it would be a huge change for a great many people who, like you, have been using it for years and years. Deprecate, maybe, but never actually remove? At a minimum, we should find where in the documentation `this.self(arguments)` is discussed, and replace it with `this.constructor`. I have no idea where that would be...

Reply:
Here, for example: http://www.qooxdoo.org/docs/#/core/classes?id=calling-another-static-method

Just enter "self(arguments)" in the search box at http://www.qooxdoo.org/docs/#